diff -ruN linux-17f135b742c4edb340afb365873c3a574f7e16cb/arch/arm/boot/dts/bcm2711-rpi-4-b.dts linux-17f135b742c4edb340afb365873c3a574f7e16cb-himax/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
--- linux-17f135b742c4edb340afb365873c3a574f7e16cb/arch/arm/boot/dts/bcm2711-rpi-4-b.dts	2024-07-29 22:00:49.058873977 +0200
+++ linux-17f135b742c4edb340afb365873c3a574f7e16cb-himax/arch/arm/boot/dts/bcm2711-rpi-4-b.dts	2025-03-31 20:35:34.131091590 +0200
@@ -56,6 +56,39 @@
 		enable-active-high;
 		gpio = <&expgpio 6 GPIO_ACTIVE_HIGH>;
 	};
+	gpio-button {
+                compatible = "gpio-keys";
+		autorepeat;
+		pinctrl-names = "default";
+		status = "okay";
+                button@0 {
+                        gpios = <&pcf8574 0 GPIO_ACTIVE_LOW>;
+                        linux,code = <116>;
+                        label = "KEY_POWER";
+                };
+                button@1 {
+                        gpios = <&pcf8574 1 GPIO_ACTIVE_LOW>;
+                        linux,code = <114>;
+                        label = "KEY_VOL_DN";
+                };
+                button@2 {
+                        gpios = <&pcf8574 2 GPIO_ACTIVE_LOW>;
+                        linux,code = <115>;
+                        label = "KEY_VOL_UP";
+                };
+ 		button@3 {
+                        gpios = <&pcf8574 3 GPIO_ACTIVE_LOW>;
+                        linux,code = <102>;
+                        label = "KEY_HOME";
+                };
+        };
+        rotary-knob {
+                       compatible = "rotary-encoder";
+                       gpios = <&pcf8574 4 0>, <&pcf8574 5 0>;
+                       linux,axis = <9>; /* REL_X */
+                       rotary-encoder,encoding = "gray";
+                       rotary-encoder,relative-axis;
+        };
 };
 
 &bt {
@@ -343,7 +376,44 @@
 &i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c1_pins>;
-	clock-frequency = <100000>;
+	//clock-frequency = <100000>;
+        status = "okay";
+        clock-frequency = <400000>;
+        i2c-scl-falling-time-ns = <50>; /* 2.5ns measured */
+        i2c-scl-rising-time-ns = <100>; /* 45ns measured */
+        hh983_serializer: serializer@18 {
+                compatible = "ti,hh983-serializer";
+                reg = <0x18>;
+                status = "okay";
+        }; 
+        hxchipset_ts@48 {
+               status = "okay";
+               compatible = "himax,hxcommon";
+               reg = <0x48>;
+               report_type = <1>;
+               interrupt-parent = <&gpio>;
+               serializer-parent = <&hh983_serializer>;
+        };
+        haptics: haptics@5a {
+               compatible = "ti,drv2605l";
+               reg = <0x5a>;
+               vbat-supply = <&sd_vcc_reg>;
+               mode = <0x02>; //DRV260X_LRA_MODE
+               library-sel = <0x07>; //DRV260X_LIB_LRA
+               vib-rated-mv = <3200>;
+               vib-overdrive-mv = <3200>;
+        };
+        pcf8574: pcf8574@38 {
+               compatible = "nxp,pcf8574";
+               reg = <0x38>;
+               gpio-controller;
+               #gpio-cells = <2>;
+               interrupt-parent = <&gpio>;
+               interrupts = <18 2>;
+               interrupt-controller;
+               #interrupt-cells = <2>;
+        };
+
 };
 
 &i2s {
